src/computation/logic/sorted-reverse-dict.ts

```diff
@@ -16,7 +16,7 @@ export class SortedReverseDict {
         const start = this.reverseDict[0].value;
         let num = 0;
 
-        while (this.reverseDict[num].value === start) {
+        while (num < this.reverseDict.length && this.reverseDict[num].value === start) {
```

Comment:
The conditional is a little unclear and complex here. Could be clarified by variables above and a more natural language conditional put into the while.

Reply:
Renamed the variables, should make it a bit more intuitive.